zingolib/src/commands.rs

```diff
@@ -242,71 +242,91 @@ impl Command for ParseAddressCommand {
-                            }
-                            Address::Tex(_) => {
-                                object! {
-                                    "status" => "success",
-                                    "chain_name" => chain_name_string,
-                                    "address_kind" => "tex",
-                                }
-                            }
-                        }
-                    },
+        if args.len() > 1 {
+            return self.help().to_string();
+        }
+        fn make_decoded_chain_pair(
+            address: &str,
+        ) -> Option<(
+            zcash_client_backend::address::Address,
+            crate::config::ChainType,
+        )> {
+            [
+                crate::config::ChainType::Mainnet,
+                crate::config::ChainType::Testnet,
+                crate::config::ChainType::Regtest(
+                    crate::config::RegtestNetwork::all_upgrades_active(),
                 ),
-                4,
-            ),
-            _ => self.help().to_string(),
+            ]
+            .iter()
+            .find_map(|chain| Address::decode(chain, address).zip(Some(*chain)))
+        }
+        if let Some((recipient_address, chain_name)) = make_decoded_chain_pair(args[0]) {
+            let chain_name_string = match chain_name {
+                crate::config::ChainType::Mainnet => "main",
+                crate::config::ChainType::Testnet => "test",
+                crate::config::ChainType::Regtest(_) => "regtest",
+            };
+            match recipient_address {
+                Address::Sapling(_) => object! {
+                    "status" => "success",
+                    "chain_name" => chain_name_string,
+                    "address_kind" => "sapling",
+                }
+                .to_string(),
+                Address::Transparent(_) => object! {
+                    "status" => "success",
+                    "chain_name" => chain_name_string,
+                    "address_kind" => "transparent",
+                }
+                .to_string(),
+                Address::Tex(_) => object! {
+                    "status" => "success",
+                    "chain_name" => chain_name_string,
+                    "address_kind" => "tex",
+                }
+                .to_string(),
+                Address::Unified(ua) => {
+                    let mut receivers_available = vec![];
+                    if ua.orchard().is_some() {
+                        receivers_available.push("orchard")
+                    }
+                    if ua.sapling().is_some() {
+                        receivers_available.push("sapling")
+                    }
+                    if ua.transparent().is_some() {
+                        receivers_available.push("transparent")
+                    }
+                    if ua.orchard().is_some()
+                        && ua.sapling().is_some()
+                        && ua.transparent().is_some()
```

Comment:
do we want to return orchard only in the case there is orchard and transparent receivers?